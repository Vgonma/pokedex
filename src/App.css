:root {
  --blue: #35548b;
}

#root {
  width: 100%;
  max-width: 1280px;
  margin: 0;
  padding: 0;
  text-align: center;
}

.capitalize {
  text-transform: capitalize;
}

.uppercase {
  text-transform: uppercase;
}

.type.normal {
  background-color: #a8a878;
}

.type.fire {
  background-color: #f08030;
}

.type.water {
  background-color: #84abf8;
}

.type.grass {
  background-color: #2cc952;
}

.type.electric {
  background-color: #f8d030;
}

.type.fighting {
  background-color: #c03028;
}

.type.poison {
  background-color: #a040a0;
}

.type.ground {
  background-color: #dfc068;
}

.type.flying {
  background-color: #a890f0;
}

.type.psychic {
  background-color: #f75887;
}

.type.bug {
  background-color: #a8b820;
}

.type.rock {
  background-color: #ab9645;
}

.type.ghost {
  background-color: #705898;
}

.type.dark {
  background-color: #705848;
}

.type.dragon {
  background-color: #7038f8;
}

.type.steel {
  background-color: #b8b8d0;
}

.type.fairy {
  background-color: #e8b1b7;
}

.head-banner {
  background-image: url('./assets/pokeball-banner-dark.jpg');
  background-repeat: no-repeat;
  background-position: center;
  height: 150px;
  display: flex;
  place-items: center;
  position: sticky;
  top: 0;
}

.page-title {
  font-size: 80px;
  margin: 1.5rem;
}

.search-form {
  width: 100%;
  max-width: 500px;
  margin-inline: 2rem;
}

.search-bar {
  width: 100%;
  max-width: 500px;
  height: 2rem;
  padding: 1rem;
}

.pokedex {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  background-color: var(--blue);
}

.pokecard {
  background-color: #5687e3;
}

.pokecard:nth-child(even) {
  background-color: #4369b2;
}

.info {
  color: white;
}

.poke-types {
  display: grid;
  grid-auto-flow: column;
}

.type {
  margin: 0;
  padding: 1rem;
}

@media screen and (max-width: 767px) {
  .page-title {
    font-size: 40px;
  }

  .head-banner {
    flex-direction: column;
  }

  .pokedex {
    grid-template-columns: 1fr;
  }

  .pokecard {
    border: 1px solid #242424;
  }

  .search-form {
    width: 90%;
  }
}
