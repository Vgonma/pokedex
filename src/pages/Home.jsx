import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { RxGear } from 'react-icons/Rx';
import { BsMic } from 'react-icons/bs';
import { getData, getNextPage } from '../redux/pokedexSlice';
import Pokedex from '../components/Pokedex';
import arrow from '../assets/up-arrow.png';

let flag = false;
let loadedAll = false;
function Home() {
  const dispatch = useDispatch();
  const { pokemon } = useSelector((state) => state);
  const [query, setQuery] = useState('');
  useEffect(() => {
    if (!flag) {
      flag = true;
      dispatch(getData(0));
    }
  });

  function getNext() {
    if (!loadedAll) {
      loadedAll = true;
      dispatch(getNextPage());
    }
  }
  return (
    <div
      id="home"
      className="page"
    >
      <div className="head-banner">
        <NavLink to="/">
          <h1 className="page-title">Pokedex</h1>
        </NavLink>
        <form className="search-form">
          <input className="search-bar" type="search" onChange={(e) => { setQuery(e.target.value); getNext(); }} placeholder="Pikachu" />
        </form>
        <div className="icons">
          <BsMic className="icon" />
          <RxGear className="icon" />
        </div>
      </div>
      <Pokedex pokemonInfo={pokemon.filter((item) => (
        query.toLowerCase() === '' ? item : item.name.toLowerCase().includes(query)
      ))}
      />
      <button className="load-all" type="button" onClick={getNext}>Load all</button>
      <a href="#home" className="top-arrow">
        <button type="button"><img src={arrow} alt="back to top" /></button>
      </a>
    </div>
  );
}

export default Home;
