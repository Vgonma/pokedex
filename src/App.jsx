import { Routes, Route } from 'react-router-dom';
import './App.css';
import Details from './pages/Details';
import Home from './pages/Home';

function App() {
  return (
    <div className="App">
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/details" element={<Details />} />
        <Route path="/*" element={<Home />} />
      </Routes>
    </div>
  );
}

export default App;
