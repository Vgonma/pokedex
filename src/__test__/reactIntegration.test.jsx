import {
  test,
  expect,
  afterEach,
} from 'vitest';
import {
  cleanup,
  render,
  screen,
} from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Home from '../pages/Home';
import App from '../App';
import store from '../redux/storeConfig';

afterEach(cleanup);

test('Should render App', () => {
  const tree = render(
    <BrowserRouter>
      <Provider store={store}>
        <App />
      </Provider>
    </BrowserRouter>,
  );
  expect(tree).toMatchSnapshot();
});

test('Should match Home snapshot', () => {
  const tree = render(<BrowserRouter><Provider store={store}><Home /></Provider></BrowserRouter>);
  expect(tree).toMatchSnapshot();
});

test('Should show head banner', () => {
  render(<BrowserRouter><Provider store={store}><Home /></Provider></BrowserRouter>);
  expect(screen.getByText(/Pokedex/i)).toBeDefined();
});
