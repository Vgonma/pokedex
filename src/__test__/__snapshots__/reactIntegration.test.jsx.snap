// Vitest Snapshot v1

exports[`Should show head banner 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="page"
        id="details"
      >
        <a
          class="back-arrow"
          href="/"
        >
          <button
            type="button"
          >
            <img
              alt="back to top"
              src="/src/assets/up-arrow.png"
            />
          </button>
        </a>
        <h1
          class="capitalize page-title"
          id="poke-details-name"
        >
          undefined #undefined
        </h1>
        <div
          class="sprite"
        >
          <img
            alt=""
            src=""
          />
        </div>
        <div
          class="details-info"
        >
          <div
            class="list-container"
          >
            <div
              class="height-weight"
            >
              <h2>
                Height
              </h2>
              <h2>
                Weight
              </h2>
            </div>
          </div>
          <div
            class="list-container"
          >
            <h2>
              Abilities
            </h2>
          </div>
          <div
            class="list-container"
          >
            <h2>
              Stats
            </h2>
          </div>
          <div
            class="list-container"
          >
            <h2>
              Games
            </h2>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="page"
      id="details"
    >
      <a
        class="back-arrow"
        href="/"
      >
        <button
          type="button"
        >
          <img
            alt="back to top"
            src="/src/assets/up-arrow.png"
          />
        </button>
      </a>
      <h1
        class="capitalize page-title"
        id="poke-details-name"
      >
        undefined #undefined
      </h1>
      <div
        class="sprite"
      >
        <img
          alt=""
          src=""
        />
      </div>
      <div
        class="details-info"
      >
        <div
          class="list-container"
        >
          <div
            class="height-weight"
          >
            <h2>
              Height
            </h2>
            <h2>
              Weight
            </h2>
          </div>
        </div>
        <div
          class="list-container"
        >
          <h2>
            Abilities
          </h2>
        </div>
        <div
          class="list-container"
        >
          <h2>
            Stats
          </h2>
        </div>
        <div
          class="list-container"
        >
          <h2>
            Games
          </h2>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
